src/main/kotlin/org/jetbrains/grammarkit/tasks/GenerateLexerTask.kt

```diff
@@ -3,25 +3,24 @@
 package org.jetbrains.grammarkit.tasks
 
 import org.apache.tools.ant.util.TeeOutputStream
-import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.tasks.*
-import org.gradle.process.ExecOperations
 import org.jetbrains.grammarkit.path
 import java.io.ByteArrayOutputStream
-import javax.inject.Inject
 
 /**
  * The `generateLexer` task generates a lexer for the given grammar.
  * The task is configured using common [org.jetbrains.grammarkit.GrammarKitPluginExtension] extension.
  */
-abstract class GenerateLexerTask @Inject constructor(
-    private val execOperations: ExecOperations,
-) : DefaultTask() {
+abstract class GenerateLexerTask : JavaExec() {
+
+    init {
+        mainClass.set("jflex.Main")
```

Comment:
TBD if this is better to put in c'tor or in `exec`

Reply:
When it was in `exec`, there was a problem with mainClass being final. In `init` it works fine